libethereum/BlockChain.cpp

```diff
@@ -226,74 +220,93 @@ bool BlockChain::open(fs::path const& _path, WithExisting _we)
+                                << m_dbPaths->extrasMinorVersionPath()
+                                << " (minor version: " << c_databaseMinorVersion << ")";
+            writeFile(m_dbPaths->extrasMinorVersionPath(), rlp(c_databaseMinorVersion));
+        }
     }
 
     try
     {
-        m_blocksDB = db::DBFactory::create(chainSubPathBlocks);
-        m_extrasDB = db::DBFactory::create(extrasSubPathExtras);
+        m_blocksDB = db::DBFactory::create(m_dbPaths->blocksPath());
+        m_extrasDB = db::DBFactory::create(m_dbPaths->extrasPath());
     }
     catch (db::DatabaseError const& ex)
     {
         // Determine which database open call failed
-        auto const dbPath = !m_blocksDB.get() ? chainSubPathBlocks : extrasSubPathExtras;
-        cerror << "Error opening database: " << dbPath;
-
+        auto const dbPath =
+            !m_blocksDB.get() ? m_dbPaths->blocksPath() : m_dbPaths->extrasPath();
         if (db::isDiskDatabase())
         {
+            LOG(m_loggerError) << "Error occurred when opening database: " << dbPath;
             db::DatabaseStatus const dbStatus = *boost::get_error_info<db::errinfo_dbStatusCode>(ex);
-            if (fs::space(path).available < 1024)
+            if (fs::space(m_dbPaths->rootPath()).available < 1024)
             {
-                cerror << "Not enough available space found on hard drive. Please free some up and "
-                        "re-run.";
+                LOG(m_loggerError)
+                    << "Not enough available space found on hard drive. Please free some up and "
+                       "re-run.";
                 BOOST_THROW_EXCEPTION(NotEnoughAvailableSpace());
             }
             else if (dbStatus == db::DatabaseStatus::Corruption)
             {
-                cerror << "Database corruption detected. Please see the exception for corruption "
-                        "details. Exception: "
+                LOG(m_loggerError)
+                    << "Database corruption detected. Please see the exception for corruption "
+                       "details. Exception: "
                     << ex.what();
                 BOOST_THROW_EXCEPTION(DatabaseCorruption());
             }
             else if (dbStatus == db::DatabaseStatus::IOError)
             {
-                cerror << "Database already open. You appear to have another instance of Aleth running.";
+                LOG(m_loggerError) << "Database already open. You appear to have another instance "
+                                      "of Aleth running.";
                 BOOST_THROW_EXCEPTION(DatabaseAlreadyOpen());
             }
         }
-        
-        cerror << "Unknown error occurred. Exception details: " << ex.what();
+
+        LOG(m_loggerError) << "Unknown error occurred when opening database. Exception details: "
+                           << ex.what();
         throw;
     }
 
     if (_we != WithExisting::Verify && !rebuildNeeded && !details(m_genesisHash))
```

Comment:
I can slightly streamline things by initializing rebuildNeeded to (_we == WithExisting::Verify) and then checking !rebuildNeeded here rather than also checking _we. This would also enable me to skip the minor version checks if rebuildNeeded == true. However, this file already has a ton of churn and this streamlining is pretty minor so I can take care of it in a future change.